### PR TITLE
Replacing ${?VAR}, ${VAR} and $VAR environment/shell variables with values in config

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

Support env var replacements from shell, refs #749

### Why are the changes needed?

This was expected to work.

### How was this patch tested?

unit tests.
Also tested on GKE.

### CODE REVIEW

File: kubectl-cloudflow/config/config.go [hunk 1]
```diff
@@ -422,14 +442,14 @@ func checkVolumeMountsReferToVolume(podsConfig *configuration.Config, containers
 	var volumesNames []string
 	var volumeMountSecretNames []string
 	if volumesConfig := podsConfig.GetConfig("pod").GetConfig(volumes); volumesConfig != nil && volumesConfig.Root().IsObject() {
-		for volumeName, _ := range volumesConfig.Root().GetObject().Items() {
+		for volumeName := range volumesConfig.Root().GetObject().Items() {
```
Comment (franciscolopezsancho) [inline kubectl-cloudflow/config/config.go:445]: well spottted

**Review by franciscolopezsancho — APPROVED** [review]
it would be nice to add something in the docs. Nicely done in any case.

**Review by ennru — APPROVED** [review]
I think this was the first Go code I hade to review.
LGTM.

**Pull request merged**